accepted/future-releases/constant-update-2018/feature-specification.md

```diff
@@ -0,0 +1,158 @@
+# Design Document for Constant Specification Update 2018.
+Author: lrn@google.com (@lrhn)
+
+## Motivation and Scope
+
+The Dart 2 and 2.1 language updates introduced a stricter type system, 
+the ability to have `assert`s in const constructors, 
+and new operators on the `int` and `bool` classes 
+(`>>>` on `int` and `|`, `&` and `^` on `bool`).
+
+The stricter type system made some constant expressions hard to write, 
+especially for users who disable implicit downcasts. 
+For that reason, we want to introduce explicit `as` casts in constant expressions.
+
+The asserts made it obvious that the existing constant expression specification
+was overly strict. 
+The `&&`, `||` and `?`/`:` operators were not specified to be short-circuiting,
+instead they always evaluated all the sub-expressions, 
+and introduced a compile-time error if any of them were wrong. 
+It made tests like `(string != null && string.length < 4)` unusable. 
+We want to fix that, to make boolean expressions more usable.
+
+The new operators on basic system types should also work in constant
+contexts, so we want to handle those as well.
+
+On top of that, the Dart 1 specification of potentially constant expressions
+had some inconsistencies where an expression was potentially constant only if
+it would actually be constant if the constructor parameters were constants
+of suitable *types*, but since the constant-ness of an expression depends on
+the actual values, not just their type, that question couldn't be answered. 
+We also want to clean that up and give a clean, statically checkable
+definition of potentially constant expressions.
+
+We do not want to change the behavior of existing constant
+or potentially constant expressions.
+
+## Design
+
+The precise defintion of the changes [has landed](https://dart-review.googlesource.com/36220)
+in the language specification. 
```

Comment:
The CL links to the commit and the commit links to the CL, so they're essentially providing the same information. However, the link https://github.com/dart-lang/sdk/commit/0808d29b4e33c303bf40524407a6e331c7a439f8 seems to be slightly more direct in terms of what the readers of this document would actually need.

Reply:
Will use that instead. Thanks.